graphql_client_codegen/src/deprecation.rs

```diff
@@ -8,22 +8,17 @@ pub enum DeprecationStatus {
 }
 
 /// The available deprecation strategies.
-#[derive(Debug, PartialEq, Clone)]
+#[derive(Debug, PartialEq, Clone, Default)]
 pub enum DeprecationStrategy {
     /// Allow use of deprecated items in queries, and say nothing.
     Allow,
     /// Fail compilation if a deprecated item is used.
+    #[default]
     Deny,
```

Comment:
This doesn't preserve the previous value. I think these clippy fixes belong in its own PR to avoid any confusion about what this PR is about.